experiment/selinux/src/selinux.rs

```diff
@@ -44,268 +95,301 @@ impl SELinux {
+    pub fn current_label(&self) -> Result<String, SELinuxError> {
+        return SELinux::read_con(self.attr_path("current").as_path());
+    }
+
+    // This function attempts to open a selinux context file, and if it fails, it tries to open another file
+    // under policy root's directory.
+    pub(crate) fn open_context_file(&mut self) -> Result<File, SELinuxError> {
+        match File::open(CONTEXT_FILE) {
+            Ok(file) => Ok(file),
+            Err(_) => {
+                let policy_path = Self::policy_root(self).ok_or_else(|| {
+                    SELinuxError::OpenContextFile("can't get policy root".to_string())
+                })?;
+                let context_on_policy_root = policy_path.join("contexts").join("lxc_contexts");
+                match File::open(context_on_policy_root) {
+                    Ok(file) => Ok(file),
+                    Err(e) => Err(SELinuxError::OpenContextFile(e.to_string())),
+                }
+            }
+        }
     }
 
-    // function similar with FormatMountLabel in go-selinux repo.
-    // format_mount_label returns a string to be used by the mount command.
-    // Using the SELinux `context` mount option.
-    // Changing labels of files on mount points with this option can never be changed.
-    // format_mount_label returns a string to be used by the mount command.
-    // The format of this string will be used to alter the labeling of the mountpoint.
-    // The string returned is suitable to be used as the options field of the mount command.
-    // If you need to have additional mount point options, you can pass them in as
-    // the first parameter. The second parameter is the label that you wish to apply
-    // to all content in the mount point.
-    pub fn format_mount_label(src: &str, mount_label: &str) -> String {
-        Self::format_mount_label_by_type(src, mount_label, "context")
+    // This returns selinux enforce path by using selinux mountpoint.
+    // The enforce path dynamically changes SELinux mode at runtime,
+    // while the config file need OS to reboot after changing the config file.
+    fn selinux_enforce_path(&mut self) -> Option<PathBuf> {
+        let selinux_mountpoint = Self::get_selinux_mountpoint(self);
+        selinux_mountpoint.map(|m| m.join("enforce"))
+    }
+
+    // enforce_mode returns the current SELinux mode Enforcing, Permissive, Disabled
+    pub fn enforce_mode(&mut self) -> SELinuxMode {
+        let mode = match Self::selinux_enforce_path(self) {
+            Some(enforce_path) => match fs::read_to_string(enforce_path) {
+                Ok(content) => content.trim().parse::<i32>().unwrap_or(-1),
+                Err(_) => -1,
+            },
+            None => -1,
+        };
+
+        match mode {
+            1 => SELinuxMode::ENFORCING,
+            0 => SELinuxMode::PERMISSIVE,
+            -1 => SELinuxMode::DISABLED,
+            _ => SELinuxMode::DISABLED,
+        }
```

Comment:
We can use From or Into instead of that.
https://doc.rust-lang.org/rust-by-example/conversion/from_into.html

Reply:
Thanks, fixed it